docs/topics/http/sessions.txt

```diff
@@ -676,6 +677,188 @@ Technical details
+
+    .. method:: encode(session_dict)
+
+        Returns the given session dictionary serialized and encoded as a string.
+
+        Encoding is performed by the session store class tied to a model class.
+
+    .. method:: save(session_key, session_dict, expire_date)
+
+        Saves session data for a provided session key, or deletes the session
+        in case data is empty.
+
+.. admonition:: Importing ``AbstractBaseSession`` and ``BaseSessionManager``
+
+    ``AbstractBaseSession`` and ``BaseSessionManager`` are importable from
+    ``django.contrib.sessions.base_session`` so that they can be imported
+    without including ``django.contrib.sessions`` in :setting:`INSTALLED_APPS`.
+
+
+Base classes for building ``SessionStore`` objects are located inside
+``django.contrib.sessions.backends``:
+
+.. class:: backends.base_db.BaseDBSessionStore
+
+    .. versionadded:: 1.9
+
+    A base class to build database-backed session engines upon.
+
+    .. classmethod:: get_session_class()
+
+        Returns a session model class to be used along with this session store.
+
+        To be overridden for a session store class to have access to a
+        customized session model class.
+
+    .. method:: create_session_instance(data)
+
+        Returns a new instance of the session model object, which represents
+        the current session state.
+
+        Gives an option to modify session model data before it goes to
+        database.
+
+.. class:: backends.base_cached_db.BaseCachedDBSessionStore
+
+    .. versionadded:: 1.9
+
+    A base class to build cached database-backed session engines upon.
+
+    Since it inherits from :class:`~backends.base_db.BaseDBSessionStore`, it
+    requires child classes to override the
+    :meth:`~backends.base_db.BaseDBSessionStore.get_session_class` classmethod.
+
+    .. attribute:: cache_key_prefix
+
+    A prefix added to a session key to build a cache key string.
+
+Example
+-------
+
+The example below shows a custom database-backed session engine that includes
+an additional database column to store an account ID (thus providing an option
+to query the database for all active sessions for an account)::
+
+    from django.contrib.sessions.backends.base_db import BaseDBSessionStore
+    from django.contrib.sessions.base_session import AbstractBaseSession
+    from django.db import models
+
+    class CustomSession(AbstractBaseSession):
+        account_id = models.IntegerField(null=True, db_index=True)
+
+        class Meta:
+            app_label = 'mysessions'
+
+        @classmethod
+        def get_session_store_class(cls):
+            return SessionStore
+
+    class SessionStore(BaseDBSessionStore):
+        @classmethod
+        def get_session_class(cls):
+            return CustomSession
+
+        def create_session_instance(self, data):
+            obj = super(SessionStore, self).create_session_instance(data)
+            try:
+                account_id = int(data.get('_auth_user_id')) or None
```

Comment:
under what conditions would `or None` be executed? _auth_user_id=0? That seems unexpected.